pkg/roachpb/api.go

```diff
@@ -771,13 +771,32 @@ func NewReverseScan(key, endKey Key) Request {
 	}
 }
 
-func (*GetRequest) flags() int                { return isRead | isTxn }
-func (*PutRequest) flags() int                { return isWrite | isTxn | isTxnWrite }
-func (*ConditionalPutRequest) flags() int     { return isRead | isWrite | isTxn | isTxnWrite }
-func (*InitPutRequest) flags() int            { return isRead | isWrite | isTxn | isTxnWrite }
-func (*IncrementRequest) flags() int          { return isRead | isWrite | isTxn | isTxnWrite }
-func (*DeleteRequest) flags() int             { return isWrite | isTxn | isTxnWrite }
-func (*DeleteRangeRequest) flags() int        { return isWrite | isTxn | isTxnWrite | isRange }
+func (*GetRequest) flags() int            { return isRead | isTxn }
+func (*PutRequest) flags() int            { return isWrite | isTxn | isTxnWrite }
+func (*ConditionalPutRequest) flags() int { return isRead | isWrite | isTxn | isTxnWrite }
+func (*InitPutRequest) flags() int        { return isRead | isWrite | isTxn | isTxnWrite }
+func (*IncrementRequest) flags() int      { return isRead | isWrite | isTxn | isTxnWrite }
+func (*DeleteRequest) flags() int         { return isWrite | isTxn | isTxnWrite }
+func (drr *DeleteRangeRequest) flags() int {
+	// DeleteRangeRequest has different properties if the "inline" flag is set.
+	// This flag indicates that the request is deleting inline MVCC values,
+	// which cannot be deleted transactionally - inline DeleteRange will thus
+	// fail if executed as part of a transaction. This alternate flag set
+	// is needed to prevent the command from being automatically wrapped into a
+	// transaction by TxnCoordSender, which can occur if the command spans
+	// multiple ranges.
+	//
+	// TODO(mrtracy): The behavior of DeleteRangeRequest with "inline" set has
+	// likely diverged enough that it should be promoted into its own command.
+	// However, it is complicated to plumb a new command through the system,
```

Comment:
I discussed this in person with @petermattis, and we both favored this solution to fix the immediate problem; I wanted to put out this solution for code review to see if there are any strong objections.  This is three lines of code, whereas the alternative could be hundreds of lines between new protos, boilerplate and tests.

In my personal opinion, compared to the overall complexity of the batching system this seems almost trivial - it actually seems like the flags() system was designed to provide exactly this sort of flexibility.